core/pkg/eval/json_evaluator.go

```diff
@@ -67,29 +69,40 @@ func (je *JSONEvaluator) GetState() (string, error) {
 func (je *JSONEvaluator) SetState(payload sync.DataSync) (map[string]interface{}, bool, error) {
 	_, span := je.jsonEvalTracer.Start(
 		context.Background(),
-		"setState",
-		trace.WithAttributes(attribute.String("feature_flag.source", payload.Source)))
+		"flagSync",
+		trace.WithAttributes(attribute.String("feature_flag.source", payload.Source)),
+		trace.WithAttributes(attribute.String("feature_flag.sync_type", payload.Type.String())))
 	defer span.End()
 
 	var newFlags Flags
+
 	err := je.configToFlags(payload.FlagData, &newFlags)
 	if err != nil {
+		span.SetStatus(codes.Error, "flagSync error")
+		span.RecordError(err)
 		return nil, false, err
 	}
 
+	var events map[string]interface{}
+	var reSync bool
+
 	switch payload.Type {
 	case sync.ALL:
-		n, resync := je.store.Merge(je.Logger, payload.Source, newFlags.Flags)
-		return n, resync, nil
+		events, reSync = je.store.Merge(je.Logger, payload.Source, newFlags.Flags)
 	case sync.ADD:
-		return je.store.Add(je.Logger, payload.Source, newFlags.Flags), false, nil
+		events = je.store.Add(je.Logger, payload.Source, newFlags.Flags)
 	case sync.UPDATE:
-		return je.store.Update(je.Logger, payload.Source, newFlags.Flags), false, nil
+		events = je.store.Update(je.Logger, payload.Source, newFlags.Flags)
 	case sync.DELETE:
-		return je.store.DeleteFlags(je.Logger, payload.Source, newFlags.Flags), true, nil
+		events = je.store.DeleteFlags(je.Logger, payload.Source, newFlags.Flags)
 	default:
 		return nil, false, fmt.Errorf("unsupported sync type: %d", payload.Type)
```

Comment:
should we also record an error for the span in this case?

Reply:
thanks for the explanation - that sounds reasonable to me :)

Reply:
Cool, I will resolve this conversation